src/main/java/com/google/devtools/build/lib/worker/WorkerKey.java

```diff
@@ -110,6 +110,15 @@ public boolean getProxied() {
     return proxied;
   }
 
+  /** Returns a user-friendly name for this worker type. */
+  public static String makeWorkerTypeName(boolean proxied) {
```

Comment:
I think it'd be better if this took a key as input instead of a boolean, given that the input seems to always come from a key.

But humm... hold on. Given that `WorkerKey` knows whether it's proxied or not already, shouldn't this be a `getWorkerTypeName` member function without arguments?

Reply:
> given that the input seems to always come from a key

Not always, the input on this line comes from `Spawn`.
https://github.com/bazelbuild/bazel/pull/10732/files#diff-4ebed75948b31f1a40fc7f782b6d31ceR127

I think since this is a static function, we don't always have the information about proxy.

Like on the same line https://github.com/bazelbuild/bazel/pull/10732/files#diff-4ebed75948b31f1a40fc7f782b6d31ceR127, we can only tell by passing argument to the static function.

If we prefer to take a key as argument, I would like to know what would the key look like. Previously you suggested `String name`, what is a `name`?

Reply:
All usages of `makeWorkerTypeName` are of the form: `WorkerKey.makeWorkerTypeName(key.getProxied())`, which is a very strange way of saying `key.getProxiedWorkerTypeName()`.

There is only once exception, which is `WorkerKey.makeWorkerTypeName(Spawns.supportsMultiplexWorkers(spawn))`. Maybe that one just needs special-casing: make the two strings constants and duplicate the conditional at that call site? Or add a `WorkerKey.getProxiedWorkerTypeName` member function that calls the static `makeWorkerTypeName`?

I guess it doesn't matter much and any of these "solutions" are actually worse. So I think this is good enough; can't see of an easy way to handle this single corner case.

Reply:
Sounds good, thanks! Let's keep it in mind, I am happy to improve this when we have better solution.